tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -4463,5 +4464,44 @@ std::optional<Value> convertSinOp(PatternRewriter& rewriter, Operation* op,
       .getResult();
 }
 
+// Lowers Sign operator to a sequence of TOSA ops.
+llvm::Optional<Value> convertSignOp(PatternRewriter& rewriter, Operation* op,
+                                    Value input, RankedTensorType output_type) {
+  if (output_type.getElementType().isa<mlir::quant::QuantizedType>()) {
+    (void)rewriter.notifyMatchFailure(op, "tfl quantization not yet supported");
+    return llvm::None;
+  }
+
+  // TOSA greater and select can both broadcast, so simply create a tensor with
+  // one element.
+  Value pos_one, neg_one, zero;
+  if (output_type.getElementType().isa<FloatType>()) {
+    pos_one = getTosaConstTensorSingleF32(rewriter, op, 1.0f);
+    neg_one = getTosaConstTensorSingleF32(rewriter, op, -1.0f);
+    zero = getTosaConstTensorSingleF32(rewriter, op, 0.0f);
+  } else {
+    pos_one = getTosaConstTensorSingleI32(rewriter, op, 1);
```

Comment:
This is hard coded to I32 types - what about I8 or I16? We should avoid hardcoding the bitwidth and instead make the implementation more generic.

Reply:
Look like quantization for Sign is not supported yet. Thus tfl.dequantize + tfl.sign.f32 + tfl.quantize is generated but not tfl.sign.i8 or tfl.sign.i16.